compiler/transf.nim

```diff
@@ -671,9 +671,8 @@ proc transformFor(c: PTransf, n: PNode): PNode =
       idNodeTablePut(newC.mapping, formal, arg)
       # XXX BUG still not correct if the arg has a side effect!
     of paComplexOpenarray:
-      let typ = newType(tySequence, formal.owner)
```

Comment:
As I wrote in your other PR, this must remain.

Reply:
As I wrote in the other PR. This line is a bug. If you are in doubt that it doesn't cover a certain case correctly, please provide it, so that I can add it as a test and therefore ensure it is working.